plugin/pki/path_import_queue.go

```diff
@@ -268,6 +270,64 @@ func (b *backend) processImportToTPP(job Job) string {
 			//TODO: Here should be renew instead of deletion
 			b.deleteCertFromQueue(job)
 		}
+
+		///
+		if (err != nil) && (cl.GetType() == endpoint.ConnectorTypeTPP) {
+			msg := err.Error()
+
+			//catch the scenario when token is expired and deleted.
+			var regex = regexp.MustCompile("(Token).*(not found)")
+
+			//validate if the error is related to a expired accces token, at this moment the only way can validate this is using the error message
+			//and verify if that message describes errors related to expired access token.
+			if (strings.Contains(msg, "\"error\":\"expired_token\"") && strings.Contains(msg, "\"error_description\":\"Access token expired\"")) || regex.MatchString(msg) {
+
+				cfg, err := b.getConfing(job.ctx, job.storage, job.policyName)
+
+				if err != nil {
+					return fmt.Sprintf("%s could not import certificate: %s\n", msg, err)
+				}
+
+				if cfg.Credentials.RefreshToken != "" {
+					msg := fmt.Sprintf("se actualizara tokem por Job id: %v ###", job.id)
+					b.Logger().Debug(msg)
+					err = synchronizedUpdateAccessToken(cfg, b, job.ctx, job.storage, job.policyName)
+
+					if err != nil {
+						return fmt.Sprintf("%s could not import certificate: %s\n", msg, err)
+					}
+
+					//everything went fine so get the new client with the new refreshed access token
+					cl, err := b.ClientVenafi(job.ctx, job.storage, job.policyName)
+					if err != nil {
+						return fmt.Sprintf("%s could not import certificate: %s\n", msg, err)
+					}
+
+					b.Logger().Debug("Reading policy configuration again")
+
+					importResp, err = cl.ImportCertificate(importReq)
+					if err != nil {
+						if errors.Is(err, verror.ServerBadDataResponce) || errors.Is(err, verror.UserDataError) {
+							//TODO: Here should be renew instead of deletion
```

Comment:
I think we should remove this TODO because it is misleading and I don't think there is anything left TODO.  The intent of this code is to keep us from retrying import of certificates that we know will always be rejected (most probably because it is denied by private key reuse policy).  So better to replace the TODO with a comment explaining that.

Reply:
fixed on recent commit